deepchecks/tabular/checks/data_integrity/feature_feature_correlation.py

```diff
@@ -81,6 +82,8 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         num_features = [f for f in dataset.numerical_features if f in df.columns]
         cat_features = [f for f in dataset.cat_features if f in df.columns]
         encoded_cat_data = df.loc[:, cat_features].apply(lambda x: pd.factorize(x)[0])
+        # NaNs are encoded as -1, replace back to NaN
+        encoded_cat_data.replace(-1, np.NaN, inplace=True)
```

Comment:
Better use pd.factorize own's `na_sentinel=None` parameter

Reply:
@noamzbr It replaces NaN with max cat + 1 (different number for each column), doesn't keep it as None, so it's better to replace all with -1 (the default) and then replace back to NaN

Reply:
Right, my bad